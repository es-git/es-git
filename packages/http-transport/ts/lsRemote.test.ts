import { promises as fs } from 'fs';
import * as sinon from 'sinon';
import lsRemote from './lsRemote';


test('error response', async () => {
  const fetch = sinon.stub();
  fetch.resolves({
    text: () => Promise.resolve('Repo not found'),
    status: 401,
    statusText: 'Unauthorized'
  });
  await expect(lsRemote('https://github.com/my/repo.git', fetch)).rejects.toThrowError('ls-remote failed with 401 Unauthorized\nRepo not found');
});

test('lsRemote', async () => {
  const fetch = sinon.stub();
  fetch.resolves({
    text: () => fs.readFile(__dirname+'/../samples/lsremote.txt', 'utf8'),
    status: 200,
    statusText: 'OK'
  });
  const result = await lsRemote('https://github.com/my/repo.git', fetch);
  expect(fetch.firstCall.args[0]).toBe('https://github.com/my/repo.git/info/refs?service=git-upload-pack');
  expect(result).toMatchSnapshot();
});
