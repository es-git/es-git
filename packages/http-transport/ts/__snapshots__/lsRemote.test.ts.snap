// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lsRemote 1`] = `
Object {
  "capabilities": Map {
    "multi_ack" => true,
    "thin-pack" => true,
    "side-band" => true,
    "side-band-64k" => true,
    "ofs-delta" => true,
    "shallow" => true,
    "deepen-since" => true,
    "deepen-not" => true,
    "deepen-relative" => true,
    "no-progress" => true,
    "include-tag" => true,
    "multi_ack_detailed" => true,
    "no-done" => true,
    "symref" => "HEAD:refs/heads/master",
    "agent" => "git/github-g7db9c1d60",
  },
  "remoteRefs": Array [
    Object {
      "hash": "40b3732b3b1fc87e625b107cd55c68cd1ba4470f",
      "name": "HEAD",
    },
    Object {
      "hash": "9c696ae2d1ffdbed22f8bf64745d3d2483f40b3c",
      "name": "refs/current",
    },
    Object {
      "hash": "0e76e6214dbba347a4f24497261a1bd71aac8347",
      "name": "refs/heads/clean",
    },
    Object {
      "hash": "5e87db001a45b8db45eba2f9c83d5a47882f1940",
      "name": "refs/heads/doc",
    },
    Object {
      "hash": "ae5b8f244fb5245fc3670a65ebb542130de5d9af",
      "name": "refs/heads/legacy",
    },
    Object {
      "hash": "40b3732b3b1fc87e625b107cd55c68cd1ba4470f",
      "name": "refs/heads/master",
    },
    Object {
      "hash": "c367dee7abfd655208ad495877acfb05f83bebbe",
      "name": "refs/heads/phase2",
    },
    Object {
      "hash": "ff690e8d8c2ab3f3ce9c1cde34503c61bf1986cc",
      "name": "refs/pull/102/head",
    },
    Object {
      "hash": "ab50767ab50c83d2af65be8e6f72baff6ce68d9f",
      "name": "refs/pull/104/head",
    },
    Object {
      "hash": "dabe3683d59f0f92ec77afd0e63a02910c53dfaa",
      "name": "refs/pull/108/head",
    },
    Object {
      "hash": "016cb40c4328f56adf1af846713132b231ea54c7",
      "name": "refs/pull/110/head",
    },
    Object {
      "hash": "f1740f50eda2d161524f80399f8e1f2a946f005a",
      "name": "refs/pull/111/head",
    },
    Object {
      "hash": "9136d27220f83f75ecf324c4b26e4020fc964eb6",
      "name": "refs/pull/112/head",
    },
    Object {
      "hash": "d4a1d6db1c059eb4f8f951e3c2835a84ad68427d",
      "name": "refs/pull/113/head",
    },
    Object {
      "hash": "01abd3194eaacb6ea57abee94d169d01e5960a9a",
      "name": "refs/pull/114/head",
    },
    Object {
      "hash": "0c92c7823b04d5c951d465c93a83dc35bb26cd95",
      "name": "refs/pull/118/head",
    },
    Object {
      "hash": "126fb88be921a80ee84ff39e98217b4981b017db",
      "name": "refs/pull/119/head",
    },
    Object {
      "hash": "221ca1b2a3059b28f036db9166d2457da2baff49",
      "name": "refs/pull/121/head",
    },
    Object {
      "hash": "57adccc9f3c194a08b91f6fd279ef6121f653381",
      "name": "refs/pull/123/head",
    },
    Object {
      "hash": "88c4573771c1d4e2ddf90232bbb576311f82a358",
      "name": "refs/pull/123/merge",
    },
    Object {
      "hash": "4380bf9b9ade4908753d1f7616d4f6ea54a1eb5d",
      "name": "refs/pull/127/head",
    },
    Object {
      "hash": "df5688681bcfb1d25f1735b7ded03a2275f5c7fb",
      "name": "refs/pull/129/head",
    },
    Object {
      "hash": "26577802397b1c865d79cf2b4f1e81a51f141e23",
      "name": "refs/pull/129/merge",
    },
    Object {
      "hash": "a455a8e283092103da87b8c9ce32c628c969edc1",
      "name": "refs/pull/135/head",
    },
    Object {
      "hash": "bdea7c8c0b043ee6c0bc0273705fc6bb9b145d71",
      "name": "refs/pull/14/head",
    },
    Object {
      "hash": "8d690a57a5aeefa2985f3fbff7fb420c896cdac9",
      "name": "refs/pull/14/merge",
    },
    Object {
      "hash": "a61ad52468e9efcc335502e5a3bb0446ec43016f",
      "name": "refs/pull/15/head",
    },
    Object {
      "hash": "e560787b85ab62db8a80e016d4372eebc2925c0f",
      "name": "refs/pull/15/merge",
    },
    Object {
      "hash": "95df3b8d82ac5df01df168c875c6f42784e04332",
      "name": "refs/pull/18/head",
    },
    Object {
      "hash": "68e300b0befd6781306450016557b74bd02603e7",
      "name": "refs/pull/18/merge",
    },
    Object {
      "hash": "bbe4beeb6f34b477ef5ffd6e3af7963093251fa0",
      "name": "refs/pull/21/head",
    },
    Object {
      "hash": "a094f210217a04b73154b61e1e59af53c3af7f2f",
      "name": "refs/pull/21/merge",
    },
    Object {
      "hash": "44f9deff485f83b9042dd02239a3665415a9d6e2",
      "name": "refs/pull/56/head",
    },
    Object {
      "hash": "b5fe74b794c24cc28a2f28e717cbb2da6828b452",
      "name": "refs/pull/56/merge",
    },
    Object {
      "hash": "bfb2575225701d7d32d30b3243bee942a49a1370",
      "name": "refs/pull/69/head",
    },
    Object {
      "hash": "f68737d67ee88bfc42b27fd2c12c62de577d9a0c",
      "name": "refs/pull/71/head",
    },
    Object {
      "hash": "5fdce0cc713390cdcc7eb24a85864d07ddc3068f",
      "name": "refs/pull/71/merge",
    },
    Object {
      "hash": "4f06a7e31f2ebc6fb4abfc6255290c6883916fe0",
      "name": "refs/pull/82/head",
    },
    Object {
      "hash": "93669de7bce48c9525ce84902df10fa7eb1ab63e",
      "name": "refs/pull/86/head",
    },
    Object {
      "hash": "552116a881f9c405a9c3699096808282968c97ae",
      "name": "refs/pull/86/merge",
    },
    Object {
      "hash": "30a3468c1c4710ff65038a16efafcd76ee909ecd",
      "name": "refs/pull/87/head",
    },
    Object {
      "hash": "2df2ba72f94915d6f273faa29c47f78602beafed",
      "name": "refs/pull/90/head",
    },
    Object {
      "hash": "8d879a826c451780e61cced91c75864e8ba4b57b",
      "name": "refs/pull/95/head",
    },
    Object {
      "hash": "92a94b28cf963e8318fde84a61b9912e15ac4eaa",
      "name": "refs/pull/97/head",
    },
    Object {
      "hash": "42d1fd4a1bbd6ff177434ad0eabbca8ecc6eeb0f",
      "name": "refs/pull/97/merge",
    },
    Object {
      "hash": "fecb23dce5ed0ccab18cb95d141e2694dc8b2763",
      "name": "refs/pull/98/head",
    },
    Object {
      "hash": "4b7fa6d4f384017f64d47f0aaf68a02fb29fe58c",
      "name": "refs/pull/99/head",
    },
    Object {
      "hash": "525af9de1e1a98b6723ce109ae13bc1455af2883",
      "name": "refs/tags/0.0.1",
    },
    Object {
      "hash": "dd63c427173810677705a1c1abca3e39bd4338e9",
      "name": "refs/tags/0.2.0",
    },
    Object {
      "hash": "3c4e5f2da0870692227febbc6ca1473d9a42c98d",
      "name": "refs/tags/0.2.1",
    },
    Object {
      "hash": "4dbeb69883264c2b87ba84adb67fa048397b3600",
      "name": "refs/tags/0.2.2",
    },
    Object {
      "hash": "e01229a1bd654109fc055a8be751c46c2d3930cb",
      "name": "refs/tags/0.2.3",
    },
    Object {
      "hash": "e6e08587d666fa04f08d82c971fc722b2c8280f5",
      "name": "refs/tags/0.2.4",
    },
    Object {
      "hash": "a3025aa5db0d2167da946d9565a610b10f196620",
      "name": "refs/tags/0.2.5",
    },
    Object {
      "hash": "7c9b204411219099d27844cbfa0706d4e685cc4f",
      "name": "refs/tags/0.3.0",
    },
    Object {
      "hash": "60478cc8107510f56a30c69ea09190ad715d1d66",
      "name": "refs/tags/0.3.1",
    },
    Object {
      "hash": "d9c04323272568da750c3bf548dff99fb4218717",
      "name": "refs/tags/0.3.2",
    },
    Object {
      "hash": "c3fe5b62c226f726cbb8a2345a69c7d78faaeb37",
      "name": "refs/tags/0.3.3",
    },
    Object {
      "hash": "0e76e6214dbba347a4f24497261a1bd71aac8347",
      "name": "refs/tags/0.4.0",
    },
    Object {
      "hash": "db477a765427f13e16b4185dd66750126f2a0dcd",
      "name": "refs/tags/0.4.1",
    },
    Object {
      "hash": "c1ef9cdf801fcbc47133b2aa73a1da1aed4231dc",
      "name": "refs/tags/0.4.2",
    },
    Object {
      "hash": "659438e9fd166caeed6172b64b20342165521508",
      "name": "refs/tags/0.5.0",
    },
    Object {
      "hash": "7857e79c9f011a6f302e4ff23d3103f376fd4fe9",
      "name": "refs/tags/0.5.1",
    },
    Object {
      "hash": "d32d42fd38f8dc01333d512a0a73d2ddd9c23ae0",
      "name": "refs/tags/0.5.2",
    },
    Object {
      "hash": "bb18d064bf79320c5719a648fd0f749b82d9ee9f",
      "name": "refs/tags/0.5.4",
    },
    Object {
      "hash": "df273f9107df5b64b2ea86f87f3bd0e5737800bd",
      "name": "refs/tags/0.6.0",
    },
    Object {
      "hash": "843c9839c7693d0d805e902111583a08a71be95c",
      "name": "refs/tags/0.6.1",
    },
    Object {
      "hash": "ae5b8f244fb5245fc3670a65ebb542130de5d9af",
      "name": "refs/tags/0.6.2",
    },
    Object {
      "hash": "96909a556176abf005b4d7deef992db6326194aa",
      "name": "refs/tags/0.7.1",
    },
    Object {
      "hash": "7e6022791f1227f1d5c6317c0e7b24974d855d99",
      "name": "refs/tags/0.7.2",
    },
    Object {
      "hash": "b5897ce2f71374b5e16ef58a5d12e4ad0ea68e39",
      "name": "refs/tags/0.7.3",
    },
    Object {
      "hash": "6dd92da330671e0b4209c9fcab6660902d0061f7",
      "name": "refs/tags/0.7.4",
    },
    Object {
      "hash": "1369459488a5b0a565f10be19b7a6fd07d4436b5",
      "name": "refs/tags/0.7.5",
    },
    Object {
      "hash": "9c696ae2d1ffdbed22f8bf64745d3d2483f40b3c",
      "name": "refs/tags/0.7.6",
    },
    Object {
      "hash": "4ca86be0bf3a5e57f53c44491838955d3201a8ff",
      "name": "refs/tags/0.7.7",
    },
    Object {
      "hash": "d78da0138969a368da74d0dfb9e3c5d88114b152",
      "name": "refs/tags/0.7.8",
    },
    Object {
      "hash": "e0947d61f70bd4cf354e74edb027ae783c5891fb",
      "name": "refs/tags/current",
    },
  ],
}
`;
